### PR TITLE
fix(gatsby-plugin-image): Better StaticImage errors

### PR DESCRIPTION
**Pull request by ascorbic**
The error when props were not resolved in StaticImage was unclear, leading to a lot of confusion, e.g. in #30143 . This PR uses the babel node path to generate a structured error, with a codeframe and link to the docs

image

### CODE REVIEW

File: packages/gatsby-plugin-image/src/node-apis/image-processing.ts [hunk 1]
```diff
@@ -122,7 +128,9 @@ export async function writeImages({
         fullPath = path.resolve(sourceDir, src)
 
         if (!fs.existsSync(fullPath)) {
-          reporter.warn(`Could not find image "${src}". Looked for ${fullPath}`)
+          reporter.warn(
+            `Could not find image "${src}" in "${filename}". Looked for ${fullPath}`
```
Comment (laurieontech) [inline packages/gatsby-plugin-image/src/node-apis/image-processing.ts:132]:
```suggestion
            `Could not find image "${src}" in "${filename}". Looked for ${fullPath}.`
```

File: packages/gatsby-plugin-image/src/node-apis/image-processing.ts [hunk 2]
```diff
@@ -80,11 +81,16 @@ export async function writeImages({
...
   createNode: Actions["createNode"]
   store: Store
+  filename: string
 }): Promise<void> {
   const promises = [...images.entries()].map(
     async ([hash, { src, ...args }]) => {
       let file: FileSystemNode | undefined
       let fullPath
+      if (!src) {
+        reporter.warn(`Missing StaticImage "src" in ${filename}`)
```
Comment (laurieontech) [inline packages/gatsby-plugin-image/src/node-apis/image-processing.ts:91]:
```suggestion
        reporter.warn(`Missing StaticImage "src" in ${filename}.`)
```

File: packages/gatsby-plugin-image/package.json [hunk 3]
```diff
@@ -71,9 +71,10 @@
     "react-dom": "^16.9.0 || ^17.0.0"
   },
   "dependencies": {
+    "@babel/code-frame": "^7.12.13",
```
Comment (laurieontech) [inline packages/gatsby-plugin-image/package.json:74]: Just want to double check that this isn't re-exported from core. I expect not, but worth asking.

**Review by laurieontech — COMMENTED** [review]
Minor edits and a question. But this is a major quality of life improvement!

File: packages/gatsby-plugin-image/package.json (hunk 3 above)
Comment (ascorbic) [reply packages/gatsby-plugin-image/package.json:74]: Doesn't appear to be

**Review by laurieontech — APPROVED** [review]
LGTM

**Pull request merged**